### PR TITLE
Clarify how to handle response bodies and add withStringBody docs

### PR DESCRIPTION
**Pull request by cleptric**
I tried to clarify the use of `withBody()`.
I also fixed some mistakes and restructured the response section a bit.

### CODE REVIEW

File: en/controllers/request-response.rst [hunk 1]
```diff
@@ -611,6 +576,68 @@ You can now use the convenience method
 :php:meth:`Cake\\Http\\Response::withLocation()` to directly set or get the
 redirect location header.
 
+Setting the Body
+----------------
+
+.. php:method:: withBody($body)
+
+To set the response body, use the `withBody()` method, which is provided by the
```
Comment (markstory) [inline en/controllers/request-response.rst]: Backticks for inline code need to be doubled up.

File: en/controllers/request-response.rst [hunk 2]
```diff
@@ -611,6 +576,68 @@ You can now use the convenience method
...
+
+    // Prior to 3.4.0 - Set the body
+    $this->response->body('My Body');
+
+Be sure that `$stream` is a :php:class:`Psr\\Http\\Message\\StreamInterface` object.
+See below on how to create a new stream.
+
+.. php:method:: withStringBody($string)
+
+In most use cases, it is simpler to set a string as the response body::
```
Comment (ravage84) [inline en/controllers/request-response.rst]: If it is simpler, why not put it first?

File: en/controllers/request-response.rst [hunk 3]
```diff
@@ -611,6 +576,68 @@ You can now use the convenience method
...
+
+    // Set a string into the body
+    $response = $response->withStringBody('My Body');
+
+    //If you want a json response
```
Comment (ravage84) [inline en/controllers/request-response.rst]: Nitpicking: Space after double slashes.

File: en/controllers/request-response.rst [hunk 4]
```diff
@@ -611,6 +576,68 @@ You can now use the convenience method
...
+    $response = $response->withType('application/json')
+        ->withBody(json_encode(['Foo' => 'bar']));
+
+.. versionadded:: 3.4.3
+    `withStringBody()` was added in 3.4.3
+
+You can also stream responses from files using diactoros streams::
```
Comment (ravage84) [inline en/controllers/request-response.rst]: Isn't diactoros a name? Using a capital letter "Diactoros", could make it more clear for non English native speakers...

File: en/controllers/request-response.rst [hunk 5]
```diff
@@ -611,6 +576,68 @@ You can now use the convenience method
...
+
+    // To stream from a file
+    use Zend\Diactoros\Stream;
+
+    $stream = new Stream('/path/to/file', 'rb');
+    $response = $response->withBody($stream);
+
+You can also stream responses from a callback using the ``CallbackStream`` this
```
Comment (ravage84) [inline en/controllers/request-response.rst]: "``CallbackStream``. This"

File: en/controllers/request-response.rst (hunk 3 above)
Comment (cleptric) [inline en/controllers/request-response.rst]: Not nitpicking. Thoroughly 👍

File: en/controllers/request-response.rst (hunk 2 above)
Comment (cleptric) [inline en/controllers/request-response.rst]: true :)

File: en/controllers/request-response.rst [hunk 6]
```diff
@@ -611,6 +576,68 @@ You can now use the convenience method
...
+.. php:method:: withStringBody($string)
+
+To set a string as the response body, do the following::
+
+    // Set a string into the body
+    $response = $response->withStringBody('My Body');
+
+    // If you want a json response
+    $response = $response->withType('application/json')
+        ->withBody(json_encode(['Foo' => 'bar']));
```
Comment (ndm2) [inline en/controllers/request-response.rst]: Should be `withStringBody` I guess...

Comment (cleptric) [inline en/controllers/request-response.rst]: Yes, thx :)

File: en/controllers/request-response.rst [hunk 7]
```diff
@@ -611,6 +576,68 @@ You can now use the convenience method
...
+
+    // Set a string into the body
+    $response = $response->withStringBody('My Body');
+
+    // If you want a json response
+    $response = $response->withType('application/json')
+        ->withStringBody(json_encode(['Foo' => 'bar']));
+
+.. versionadded:: 3.4.3
+    `withStringBody()` was added in 3.4.3
```
Comment (markstory) [inline en/controllers/request-response.rst]: Another place backticks need to be doubled.

File: en/controllers/request-response.rst [hunk 8]
```diff
@@ -611,6 +576,68 @@ You can now use the convenience method
...
+
+To set the response body, use the ``withBody()`` method, which is provided by the
+:php:class:`Zend\\Diactoros\\MessageTrait`::
+
+    $response = $response->withBody($stream);
+
+    // Prior to 3.4.0 - Set the body
+    $this->response->body('My Body');
+
+Be sure that `$stream` is a :php:class:`Psr\\Http\\Message\\StreamInterface` object.
```
Comment (markstory) [inline en/controllers/request-response.rst]: Here too.

**Pull request merged**